### ISSUE TITLE
Remove realization $schema->slug. Add dependence eloquent-sluggable.

### ISSUE THREAD

**Issue by MaximKiselev** [cruddy/cruddy#116]
Как вы отнесете к такому подходу? $schema-slug работает коряво.

**Comment by lazychaser** [comment]
Почему коряво-то?

**Comment by MaximKiselev** [comment]
Я пробовал с русским текстом не работает (chars тоже смотрел - странное какое то решение). Нельзя поставить разделители. Плагин я подключил который выше написал. Не понял как настраивать на update, те если нужно обновлять slug при обновлении поста. Это я про schema->slug... Просто наверно надо слуг расширить, либо просто использовать зависимость а метод чтобы брал уже функционал той библиотеки.

**Comment by lazychaser** [comment]
​​
В стандартной реализации используется str_slug, который преобразует также
русские символы в транслит. Чтобы обновить slug, достаточно вписать в поле
любой текст (или нажать на кнопку синхрозиации с другим полем) и он при
сохранении преобразуется в slug

5 января 2016 г., 10:14 пользователь Maxim Kiselev <notifications@github.com
> написал:

> Я пробовал с русским текстом не работает (chars тоже смотрел - странное
> какое то решение). Нельзя поставить разделители. Плагин я подключил который
> выше написал. Не понял как настраивать на update, те если нужно обновлять
> slug при обновлении поста. Это я про schema->slug... Просто наверно надо
> слуг расширить, либо просто использовать зависимость а метод чтобы брал уже
> функционал той библиотеки.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cruddy/cruddy/issues/116#issuecomment-168921115>.
>



-- 
Best regards, Alexander Kalnoy.

**Comment by MaximKiselev** [comment]
Да, пробовал. Проблема не в этом, проблема в том, что нет разделителей.Транслитировать он тоже отказался. Редактировать каждый товар или пост таким методом не удобно.

**Comment by lazychaser** [comment]
Обновите cruddy до актуальной версии 0.5.5

5 января 2016 г., 10:25 пользователь Maxim Kiselev <notifications@github.com
> написал:

> Да, пробовал. Проблема не в этом, проблема в том, что нет
> разделителей.Транслитировать он тоже отказался. Редактировать каждый товар
> или пост таким методом не удобно.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cruddy/cruddy/issues/116#issuecomment-168922627>.
>



-- 
Best regards, Alexander Kalnoy.

**Comment by MaximKiselev** [comment]
Еще нашел несколько странных вещей. Спамить не хочу, если действительно подтвердится. То можно в исуе вынести.
1) Entity-Model cвязь.
Пытался связать 2 модели по Foreign Key. В Entity $schema->relates('foo', 'foo');
В модели связь по BelongTo('App\Model') - так работает, с публичным методом связи. А если мы ставим внешний ключ вроде $this->belongTo('App\Model', local_field (foreign key)) - так не работает (Название поля связи  без ID. просто category, например в 1 случае работает с ID, а во втором не работает.) При этом в запросе при сохранении идет пустота, заместо ключа.
2) Ошибка UI. $schema->image('')->many()
`app.src.coffee:1422 Uncaught TypeError: c.push is not a function
`
**  "kalnoy/cruddy": "^0.5.5",** Моя версия с кампостера.

**Comment by lazychaser** [comment]
Не очень понял я по пункту 1.
По пункту 2 посмотрю.

**Comment by MaximKiselev** [comment]
@lazychaser  хех, да порой меня сложно понять.
Сейчас поясню.
Есть 2 модели. Мы связываем через Entity Relationshiops. ($schema->relates()). Все норм связалось, те ищет, фильтрует. Но когда мы пытаемся это все дело сохранить идет пустой ключ.
Изначально тестил , внешний ключ в таблице без id. Например,
Models:
App\Article   
App\Category
Схема статьи
name | content | category(внешний ключ)
теперь в моделях.
```
class Article extends Model {
   
    protected $table = 'articles';
    public function category()
    {
        return  $this->belongsTo('App\Category');
    }
}

class Category extends Model { /*** ***/}

```
Так вот при таком раскладе, даже с указанием поля внешнего ключа. Он его не находит. Если ставим в поле ID, то все работает.

**Comment by lazychaser** [comment]
Все равно ничего не понятно. В любом случае cruddy использует встроенные средства laravel, так что это скорее всего недопонимание того, как все работает.
Второй пункт пофиксен в 0.5.6@dev

**Comment by MaximKiselev** [comment]
Ясно, спс за фикс с coffeescript. C Праздниками!

**Comment by lazychaser** [comment]
И вас))

2016-01-05 14:32 GMT+03:00 Maxim Kiselev <notifications@github.com>:

> Ясно, спс за фикс с coffeescript. C Праздниками!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cruddy/cruddy/issues/116#issuecomment-168977176>.
>



-- 
Best regards, Alexander Kalnoy.

**Comment by MaximKiselev** [comment]
дропдаун не работает.... все позиции в undefined items: `[{id: 1, body: "sss"}, {id: 2, body: "sss"}, {id: 3, body: "ssss"}]` атрибут задан.
Версия Cruddy dev-master e7afe75
Версия laravel 5.2.10
Думаю опять дела с конфескриптом.

**Comment by lazychaser** [comment]
​Тебе нужно их опубликовать
https://github.com/cruddy/cruddy/wiki/Getting-started#installation и не
забыть добавить в composer этот скрипт

14 января 2016 г., 3:38 пользователь Maxim Kiselev <notifications@github.com
> написал:

> дропдаун не работает.... все позиции в undefined items: [{id: 1, body:
> "sss"}, {id: 2, body: "sss"}, {id: 3, body: "ssss"}] атрибут задан.
> Версия Cruddy dev-master e7afe75
> <https://github.com/cruddy/cruddy/commit/e7afe757b9abb1e10e4b6f1d52d8dd0e200a3a33>
> Версия laravel 5.2.10
> Думаю опять дела с конфескриптом.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cruddy/cruddy/issues/116#issuecomment-171484734>.
>



-- 
Best regards, Alexander Kalnoy.

**Comment by MaximKiselev** [comment]
да, спасибо,в кампосте паблиш не стоял. еще такой вопрос: можно вывести свой шаблон в дашборде(для статистики/настроек)?

**Comment by lazychaser** [comment]
не понял, что значит "свой шаблон"? Ты имеешь в виду вывести инфу в шаблоне cruddy? Для этого есть шаблон `cruddy::custom`, в котором тебе нужно выводить инфу в секциях `title` и `body`

**Comment by MaximKiselev** [comment]
ну те вывести не crud а страницу с кастомными формами или entity.cruddy::custom - это поможет?
